### DISCUSSION TITLE
How to detect if a helm chart is using OCI

### DISCUSSION THREAD

**Discussion by patst** [argoproj/argo-cd#8277] [Q&A]
We are using the Argo `Application` CRD to define our applications. 
We use helm as deployment manager and reference the charts out of an Azure Container registry.

We would like to use RenovateBot (https://github.com/renovatebot/renovate) to automatically update the chart versions if there is a newer image in the container registry.

There is a RenovateBot manager for argocd (https://docs.renovatebot.com/modules/manager/argocd/ ) which is able to update helm charts referenced out of git repositories and helm chart repositories.

I was trying to implement a feature doing the update for OCI helm charts as well, see https://github.com/renovatebot/renovate/issues/13699 . I made the assumption, that helm `repoURL`s not containing a protocol (e.g `https`) are OCI reference. But I am not sure if this assumption is correct and the documentation does not really help.

Does somebody know if the assumption is okay to make or is there a better way?

**Comment by todaywasawesome** [comment]
I am not sure that's a safe assumption. You might ping @scottrigby as he worked on this feature for Helm. Or ask in the [Helm slack](https://cloud-native.slack.com/archives/CGF207RB5). I don't think this is a great issue for Argo CD.

**Reply by patst** [reply]
@todaywasawesome  thanks for your reply.

To be honest I think ArgoCD is behaving a little bit different to plain helm from what I saw.
Taking a look at the helm docs there is always a `oci://` prefix for OCI charts (https://helm.sh/docs/topics/registries/#other-subcommands ). 
ArgoCD on the other hand assumes it is a OCI helm chart if no `https://` is in the url (see the test case here: https://github.com/argoproj/argo-cd/blob/master/util/helm/client_test.go#L118 )

The `oci://` seems to be trimmed from the `repoURL` in some places(https://github.com/argoproj/argo-cd/blob/master/reposerver/repository/repository.go#L515 ). 

If I configure a application with `repoURL=oci://<containerregname>.azurecr.io` there is an error during argocd sync like this:

```
ComparisonError: rpc error: code = Unknown desc = `helm pull --destination /tmp/helm500916245 --version 1.9.1 --repo oci://<containerregname>.azurecr.io helm/argo/argocd-applicationset` failed exit status 1: Error: looks like "oci://<containerregname>.azurecr.io" is not a valid chart repository or cannot be reached: object required
```

I guess because it is taking the `else` branch here and not recognising that it is an OCI chart which shouldn't have an `--repo` parameter during pull (https://github.com/argoproj/argo-cd/blob/master/util/helm/client.go#L164 )
